### PR TITLE
chore: Stop processing team events (WPB-4538)

### PR DESCRIPTION
**Pull request by thisisamir98**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4538" title="WPB-4538" target="_blank">EpicWPB-4538</a>  [Platform] Stop processing team events
  </td></table>
  <br />
 

The following team events are deprecated on backend and we need to skip processing them, expect the `team.member-leave` which we need to keep for backward compatibility.

team.member-join
team.member-leave
team.update
team.member-update

### CODE REVIEW

File: test/helper/EventGenerator.ts [hunk 1]
```diff
@@ -68,6 +72,24 @@ export function createReactionEvent(targetMessageId: string, reaction: string =
...
 
+export function createMemberLeaveEvent(conversationId: string, userIds: string[]): MemberLeaveEvent {
+  const conversationQualifiedId = {id: conversationId, domain: ''};
+
+  return {
+    conversation: conversationId,
+    qualified_conversation: conversationQualifiedId,
+    data: {
+      qualified_user_ids: userIds.map(userId => ({id: userId, domain: ''})),
+      reason: MemberLeaveReason.USER_DELETED,
```
Comment (atomrc) [inline test/helper/EventGenerator.ts:83]: I think we are missing the part that displays the `x was removed from the team` when this value is set on the event. Am I missing something?

Comment (thisisamir98) [reply test/helper/EventGenerator.ts:83]: fixed in https://github.com/wireapp/wire-webapp/pull/16471/commits/3edd9bc30e76f711cfa6ee3274e035c0e8e0ef50

File: yarn.lock [hunk 2]
```diff
@@ -8771,12 +8771,12 @@ __metadata:
   linkType: hard
 
 "follow-redirects@npm:^1.15.4":
-  version: 1.15.4
-  resolution: "follow-redirects@npm:1.15.4"
+  version: 1.15.5
```
Comment (atomrc) [inline yarn.lock:8819]: This change is suspicious 🤔

File: src/script/event/preprocessor/EventStorageMiddleware/EventStorageMiddleware.ts [hunk 3]
```diff
@@ -73,9 +80,37 @@ export class EventStorageMiddleware implements EventMiddleware {
   }
 
   private validateEvent(event: HandledEvents, duplicateEvent?: EventRecord) {
+    if (event.type === CONVERSATION_EVENT.MEMBER_LEAVE) {
```
Comment (atomrc) [inline src/script/event/preprocessor/EventStorageMiddleware/EventStorageMiddleware.ts:83]: Let's add some comment for future reference

```suggestion
    if (event.type === CONVERSATION_EVENT.MEMBER_LEAVE) {
    /* When we receive a `member-leave` event, we should check that the user is actually still part of the conversation before forwarding the event. If the user is already not part of the conversation, then we can throw a validation error (that means the user was already removed by another member-leave event) */
    
```

**Review by atomrc — APPROVED** [review]

**Pull request merged**